ulx/src/main/java/com/uplink/ulx/bridge/Bridge.java

```diff
@@ -828,6 +827,14 @@ public void sendInternet(URL url, JSONObject jsonObject, int test) {
         getNetworkController().sendInternet(url, jsonObject, test);
     }
 
+    public void setInternetConnectionCallback(NetworkController.InternetConnectivityDelegate delegate) {
+        // set the delegate
+        getNetworkController().setInternetConnectivityDelegate(delegate);
+
+        // issue a first request to get the initial state of internet connectivity
+        getNetworkController().checkForInternetConnection();
```

Comment:
This will initiate connection check indeed, but it won't deliver any notifications. In reality, it appears that the notification will be issued soon after `NetworkStateListener` is registered in case there is internet available, but nothing will happen it's not. So we need either to force initial state update from `NetworkStateListener` or to make `NetworkController` handle internet availability information it gets in `checkForInternetConnection()`

Reply:
But considering that we assume no internet connection from the beginning in `BleDriver`, perhaps we don't need this at all

Reply:
We do need to check for internet connection initially, in the case that the device already has internet, to change the default from false to true. I wanted this to work asynchronously and that function `checkForInternetConnection()` is missing the part to notify the internetConnectivityDelegate. Will add that.
So it would start as false, then checkForInternetConnection() is called and will return true through the delegate.